src/osd/scrubber/scrub_reservations.h

```diff
@@ -64,6 +81,13 @@ class ReplicaReservations {
   /// for logs, and for detecting slow peers
   ScrubTimePoint m_last_request_sent_at;
 
+  /// a token to uniquely identify the last reservation request sent.
+  /// The response received will be checked against this nonce, to detect
+  /// stale responses.
+  /// Starts at 1, and is incremented for each request sent. '0' is reserved
+  /// for legacy messages.
```

Comment:
This comment doesn't really explain why the mechanism is necessary or sufficient.

```
/**
 * m_last_request_sent_nonce
 *
 * The primary is permitted to cancel outstanding reservation requests without
 * waiting for the pending response from the replica.  Thus, we may, in general,
 * see responses from prior reservation attempts that we need to ignore.  Each
 * reservation request is therefore associated with a nonce incremented within
 * an interval with each reservation request.  Any response with a non-matching
 * nonce must be from a reservation request we canceled.  Note that this check
 * occurs after validating that the message is from the current interval, so reusing
 * nonces between intervals is safe.
 *
 * 0 is a special value used to indicate that the sender did not include a nonce due
 * to not being a sufficiently recent version.
 */
```